src/logmessage.cpp

```diff
@@ -122,7 +122,11 @@ namespace g3 {
 
 
    std::string LogMessage::timestamp(const std::string& time_look) const {
-      return  localtime_formatted(_timestamp, time_look);
+      std::ostringstream stringStream;
+
+      stringStream << _timestamp.tv_nsec;
+
+      return g3::localtime_formatted(_timestamp.tv_sec, (time_look.empty() ? g3::internal::date_formatted + " " + g3::internal::time_formatted : time_look)) + "|" + stringStream.str() + "|";
```

Comment:
Indeed it is getting better.
I just committed few changes to wire timespec timestamp with formatting part; now put_time receives additional parameter const long nsec which can be used for generating formatted string with fractional sec part (it is still missing though).
Few questions:
- I created one more localtime_formatted function with different signature that accepts timespec argument. Do you want to keep the old one with time_t parameter?
- I think systemtime_now should be changed to return timespec instead of time_t. Do you agree?

If you agree to remove old version of localtime_formatted and change systemtime_now then these will change g3 API as sinks need to be changed accordingly. So it is up to you ;)
- Even though I can already get access to fractional part from sinks like following:
`logMessage.timestamp({"%Y/%m/%d %H:%M:%S"}) + "." + logMessage._timestamp.tv_nsec`
I plan to enhance put_time to support following formatting options: %f - sec fractional part (default nanosec precision); %f[1...6] - sec fraction part (precision is as defined by the digit after %f)

Do you agree?